packages/server/lib/config.ts

```diff
@@ -138,7 +138,11 @@ export function mergeDefaults (
     config.baseUrl = url.replace(/\/\/+$/, '/')
   }
 
-  const defaultsForRuntime = configUtils.getDefaultValues(options)
+  let defaultsForRuntime = configUtils.getDefaultValues(options)
+
+  // As we normalize the config based on the selected testing type, we need
+  // to do the same with the default values to resolve those correctly
+  defaultsForRuntime = { ...defaultsForRuntime, ...defaultsForRuntime[options.testingType] }
```

Comment:
Interesting - kind of surprised we need to do this, I thought this was a UI only bug - so was the bug more deeper in the system, too?

The config stuff is so spread out all over the place, probably why we have some many bugs around it :(

Reply:
+1 all configuration things should prob be happening in the config package to keep this logic in one place and so we can make it easily testable.